app/controlplane/internal/service/attestation.go

```diff
@@ -186,34 +189,75 @@ func (s *AttestationService) Init(ctx context.Context, req *cpAPI.AttestationSer
 }
 
 func (s *AttestationService) Store(ctx context.Context, req *cpAPI.AttestationServiceStoreRequest) (*cpAPI.AttestationServiceStoreResponse, error) {
+	var envelope dsse.Envelope
 	robotAccount := usercontext.CurrentRobotAccount(ctx)
 	if robotAccount == nil {
 		return nil, errors.NotFound("not found", "robot account not found")
 	}
 
-	// This will make sure the provided workflowRunID belongs to the org encoded in the robot account
-	wf, err := s.findWorkflowFromTokenOrNameOrRunID(ctx, robotAccount.OrgID, "", "", req.WorkflowRunId)
+	// Try unmarshaling a bundle first, falling back to plain dsse envelopes
+	var bundle protobundle.Bundle
+	if req.GetBundle() != nil {
+		if err := protojson.Unmarshal(req.GetBundle(), &bundle); err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		}
+		envelope = *envelopeFromBundle(&bundle)
+	} else {
+		// trying with envelope instead
+		if err := json.Unmarshal(req.GetAttestation(), &envelope); err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		}
+	}
```

Comment:
should we check that envelope is not nil and return otherwise?

Reply:
Done